src/app/clusters/occupancy-sensor-server/README.md

```diff
@@ -0,0 +1,170 @@
+virtual methods.
+
+```cpp
+#include "app/clusters/occupancy-sensor-server/OccupancySensingCluster.h"
+
+class MyOccupancyDelegate : public chip::app::Clusters::OccupancySensingDelegate
+{
+public:
+    void OnOccupancyChanged(bool occupied) override
+    {
+        // Your logic to react to an occupancy change
+    }
+
+    void OnHoldTimeChanged(uint16_t holdTime) override
+    {
+        // Your logic to react to a hold time change
+    }
+};
+```
+
+### 2. Instantiate Delegates and Cluster
+
+Instantiate your optional delegate, a timer delegate (if using the Hold Time
+feature), and the `OccupancySensingCluster` itself for each required endpoint.
+Using `RegisteredServerCluster` simplifies registration.
+
+The `HoldTime` feature and its related attributes are optional. To enable them,
+call the `.WithHoldTime()` method on the `Config` object. If it is not called,
+these attributes will be disabled.
+
+```cpp
+#include "app/DefaultTimerDelegate.h"
+#include "app/clusters/occupancy-sensor-server/OccupancySensingCluster.h"
+#include "app/server-cluster/ServerClusterInterfaceRegistry.h"
+
+// In a .cpp file
+MyOccupancyDelegate gMyOccupancyDelegate;
+DefaultTimerDelegate gTimerDelegate;
+
+constexpr chip::app::Clusters::OccupancySensing::Structs::HoldTimeLimitsStruct::Type kHoldTimeLimits = {
+    .holdTimeMin = 1, .holdTimeMax = 300, .holdTimeDefault = 60
+};
+
+chip::app::RegisteredServerCluster<chip::app::Clusters::OccupancySensingCluster> gOccupancyCluster(
+    chip::app::Clusters::OccupancySensingCluster::Config(kYourEndpointId)
+        .WithFeatures(chip::app::Clusters::OccupancySensing::Feature::kPassiveInfrared)
+        .WithHoldTime(60, kHoldTimeLimits, gTimerDelegate) // Enable HoldTime feature
+        .WithDelegate(gMyOccupancyDelegate)                // Attach optional delegate
+);
+```
+
+### 3. Register the Cluster
+
+In your application's initialization sequence, register the cluster instance
+with either the `CodegenDataModelProvider` (legacy) or the
+`CodeDrivenDataModelProvider`.
+
+```cpp
+#include "data-model-providers/codegen/CodegenDataModelProvider.h"
+
+void ApplicationInit()
+{
+    // ... other initializations
+    CHIP_ERROR err = chip::app::CodegenDataModelProvider::Instance().Registry().Register(gOccupancyCluster.Registration());
+    // ...
+}
+```
+
+### 4. Notify the Cluster of Occupancy Changes
+
+Your application is responsible for monitoring the physical sensor. When the
+sensor state changes, call the `SetOccupancy()` method on the cluster instance.
+This must be called within the Matter context. Use `ScheduleWork` if calling
+from a separate application thread.
+
+```cpp
+void MySensorHardwareCallback(bool isOccupied)
+{
+    // Get the cluster instance (e.g., via the RegisteredServerCluster object)
+    chip::app::Clusters::OccupancySensingCluster * cluster = gOccupancyCluster.Get();
+    if (cluster)
+    {
+        cluster->SetOccupancy(isOccupied);
+    }
+}
+```
+
+## Legacy Ember-Style Usage
```

Comment:
I think we don't need to mention Ember, the ember- callbacks have outlived their time in the sdk, just legacy should be fine.